net-analyzer/ossec-hids/ossec-hids-3.6.0.ebuild

```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DESCRIPTION="Open Source Host-based Intrusion Detection System"
+HOMEPAGE="https://www.ossec.net/"
+SRC_URI="https://github.com/ossec/ossec-hids/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="agent hybrid local mysql postgres server sqlite"
```

Comment:
While a typical setup will have more agents than managers, I don't feel comfortable enabling a default flag. Setting up a HIDS requires planning and forethought. Having users decide on a setup model by specifying a single USE flag is not too much to ask.

Reply:
Arch (AUR PKGBUILD) seems to build 'server', 
FreeBSD seems to enable 'local' by default but is obviously changeable similarily to Gentoo, 
OpenBSD seems to enable 'server' by default, and
Alpine Linux seems to, from what I can tell, build it with both 'server' and 'agent'. Are you sure those two can't be enabled simultaneously? 

Anyway judging from these I'd put 'server' as default.

Reply:
> Alpine Linux seems to, from what I can tell, build it with both 'server' and 'agent'. Are you sure those two can't be enabled simultaneously?

Based on [the documentation](https://www.ossec.net/docs/docs/development/build/makefile.html), "`TARGET` Defines the type of installation to build" and can have one of the listed values.

> Anyway judging from these I'd put 'server' as default.

I do not believe it wise to take this decision away from the user who must deal with the outcome. A user who cannot figure out what installation type is best for his particular needs should talk to an expert.

While I cannot prevent others to change this package after my pull request is processed, I don't want to sign off on what I consider to be a bad idea. After all, this is not about a technical problem in the ebuild, like a missing dependency or restriction.

I therefore respectfully disagree with your proposal to set a default target, which I hope you can understand.

Reply:
I feel like no matter what I'm gonna say we won't agree on this one :)

This is how alpine builds it, 
https://git.alpinelinux.org/aports/tree/testing/ossec-hids/APKBUILD?id=b72493a1d2b9abeddc3a756f78dd25e8028b2a00#n34
https://git.alpinelinux.org/aports/tree/testing/ossec-hids/APKBUILD?id=b72493a1d2b9abeddc3a756f78dd25e8028b2a00#n68

it might be interesting / worthwhile to test out similar method.

Reply:
We can agree to disagree. ;-)

As for the Alpine build: That's a loop over "make clean; make TARGET=$x" with x in [agent, server]. That's two builds, and as far as I understand OSSEC architecture it does not make sense to do it this way. Either a machine is collecting other machines' reports (server) or it is reporting (agent), not both.